docs/design/sql-review-source-code-tour.snb.md

```diff
@@ -0,0 +1,107 @@
+
+The SQL review rule contains Type, Level and rule dependent Payload.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/sql_review.go?L143-150
+
+The Type specifies the rule. The Level defines the error level for this rule, Error, Warning or Disabled.
+
+## The SQL Advisor
+
+We implement each rule as a SQL advisor in Bytebase. The SQL advisor is SQL dialect dependent, because we should use different SQL parser for each SQL dialect.
+
+The SQL advisor is a plugin in Bytebase backend. The advisor needs to implement a Check function.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/advisor.go?L201-204
+
+Because of the plugin architecture, each SQL advisor needs to call Register to register itself.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/advisor.go?L211-231
+
+Also, we have a tuple (SQL review type, SQL dialect) to SQL advisor mapping.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/sql_review.go?L323-331
+
+We have implemented some MySQL dialects (MySQL and TiDB) and PostgreSQL advisors. Here, we introduce the MySQL table naming convention advisor as an example. 
+
+### The MySQL TableNamingConvention Advisor
+
+This advisor checks the table naming convention. Let’s implement the Check function.
+
+First, we should parse the SQL statement.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/mysql/advisor_naming_table.go?L27-32
+
+Then, we need to get the error level and payload.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/mysql/advisor_naming_table.go?L34-41
+
+The TableNamingConvention needs the naming format and maximum length which stores in the rule payload.
+
+Next, we should check the new table names in SQL statement. If any of them break this rule, we should add an advice. To make this purpose, we should visit the AST(Abstract Syntax Tree) to find out all new table names.
+
+So we need to focus on CREATE TABLE, ALTER TABLE RENAME and RENAME TABLE statements.
+
+We use TiDB parser as the MySQL dialect parser. And we implement the ast.Visitor interface to visit the TiDB parser AST.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/mysql/advisor_naming_table.go?L71-92
+
+The advisor checks all new table names based on the rule.
+
+https://sourcegraph.com/github.com/bytebase/bytebase@72e8995/-/blob/plugin/advisor/mysql/advisor_naming_table.go?L94-113
+
+
+## The Catalog
+
+The catalog is the schema information for a database. For some rules, we need some information from the catalog. e.g. if we have a SQL `ALTER TABLE t RENAME INDEX uk to uk_name;` and want to check the columns in this index, we can not get this information from this SQL statement. So we need a catalog in this case.
+
+
+## How to Implement a SQL Advisor
```

Comment:
We can also link the sql-advisor README.md to here to avoid duplication.

Reply:
I think you mean SQL advisor generator README.md